base/bitarray.jl

```diff
@@ -15,6 +15,12 @@ over `Array{Bool, N}` and allowing some operations to work on 64 values at once.
 By default, Julia returns `BitArrays` from [broadcasting](@ref Broadcasting) operations
 that generate boolean elements (including dotted-comparisons like `.==`) as well as from
 the functions [`trues`](@ref) and [`falses`](@ref).
+
+Note that extra care has to be taken with regards to thread safety when
+mutating a `BitArray` due to its packed storage format. In particular, `A[i]`
+and `A[i+n]` share storage if `n` is small (that is, if `n <= 64 - mod1(i, 64)`) so any
```

Comment:
You may want `0 <= n`...

Also, I really don't find this way of expressing the condition easy to follow..... and apparently it's hard to get right either.....

I would just say `(i - 1) % 64 == (j - 1) % 64`. This way, it should be obvious that the condition is symmetric and has something to do with 64bit blocks.

Reply:
You want `(i - 1) & 64 == (j - 1) & 64` if we're going to talk about indices `i` and `j`.  I like the `+n` form just because it emphasizes the "closeness" of the indices and read a bit simpler to me — it's essentially indices within 64 of eachother that are dangerous.

Reply:
The fact that we can't get this right after many iterations suggests that it's neither easily readable nor writeable.  It's `& 0xFFFF_FFFF_FFFF_FFF0` — or simpler, [`>> 6`](https://github.com/JuliaLang/julia/blob/2fcb1b16fc734be41ebf348ddfe37809b081f027/base/bitarray.jl#L99).  Mod isn't right — that identifies the bit within a chunk. We want the chunk index.

I think.  Right?

Reply:
> The fact that we can't get this right after many iterations suggests that it's neither easily readable nor writeable.

Well, not writable maybe, but readable. It's `÷ 64`.